dvc/repo/experiments/remove.py

```diff
@@ -15,101 +34,148 @@
 @scm_context
 def remove(
     repo,
-    exp_names=None,
-    queue=False,
-    clear_all=False,
-    remote=None,
-    **kwargs,
-):
-    if not any([exp_names, queue, clear_all]):
+    exp_names: Union[None, str, List[str]] = None,
+    rev: Optional[str] = None,
+    all_commits: bool = False,
+    num: int = 1,
+    queue: bool = False,
+    git_remote: Optional[str] = None,
+) -> int:
+    if not any([exp_names, queue, all_commits, rev]):
         return 0
 
     removed = 0
     if queue:
         removed += _clear_stash(repo)
-    if clear_all:
-        removed += _clear_all(repo)
+    if all_commits:
+        removed += _clear_all_commits(repo.scm, git_remote)
+        return removed
 
+    commit_ref_set: Set[ExpRefInfo] = set()
+    queued_ref_set: Set[int] = set()
     if exp_names:
-        removed += _remove_exp_by_names(repo, remote, exp_names)
+        _resolve_exp_by_name(
+            repo, exp_names, commit_ref_set, queued_ref_set, git_remote
+        )
+
+    if rev:
+        _resolve_exp_by_baseline(repo, rev, num, commit_ref_set, git_remote)
+
+    if commit_ref_set:
+        removed += _remove_commited_exps(repo.scm, commit_ref_set, git_remote)
+
+    if queued_ref_set:
+        removed += _remove_queued_exps(repo, queued_ref_set)
+
     return removed
 
 
+def _resolve_exp_by_name(
+    repo,
+    exp_names: Union[str, List[str]],
+    commit_ref_set: Set["ExpRefInfo"],
+    queued_ref_set: Set[int],
+    git_remote: Optional[str],
+):
+    remained = set()
+    if isinstance(exp_names, str):
+        exp_names = [exp_names]
+
+    exp_ref_dict = resolve_name(repo.scm, exp_names, git_remote)
+    for exp_name, exp_ref in exp_ref_dict.items():
+        if exp_ref is None:
+            remained.add(exp_name)
+        else:
+            commit_ref_set.add(exp_ref)
+
+    if not git_remote:
+        stash_index_dict = _get_queued_index_by_names(repo, remained)
+        for exp_name, stash_index in stash_index_dict.items():
+            if stash_index is not None:
+                queued_ref_set.add(stash_index)
+                remained.remove(exp_name)
+
+    if remained:
+        raise InvalidArgumentError(
+            "'{}' is not a valid experiment".format(";".join(remained))
```

Comment:
Sounds like each `remained` are experiments but they're not valid. Is that the case? I suspect we mean that they're not valid experiment names. If so we can say so (`not a valid experiment name`) or something like `could not find experiment {}`.